2024-06-29 13:00:50,167 Exception on /login [GET] [in /home/mokeira/.local/lib/python3.8/site-packages/flask/app.py:828]
Traceback (most recent call last):
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/routes.py", line 111, in login
    return render_template('login.html', title='Login', form=form)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 150, in render_template
    return _render(app, template, context)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 131, in _render
    rv = template.render(context)
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/templates/login.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "/mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/templates/base.html", line 68, in top-level template code
    {% block content %}
  File "/mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/templates/login.html", line 59, in block 'content'
    <a href="{{ url_for('Request Password Reset')}}"> Reset Here</a>
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 1074, in url_for
    return self.handle_url_build_error(error, endpoint, values)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 1063, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
  File "/home/mokeira/.local/lib/python3.8/site-packages/werkzeug/routing/map.py", line 919, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'Request Password Reset'. Did you mean 'static' instead?
2024-06-29 13:19:26,443 Tinker Full-Stack App [in /mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/__init__.py:51]
2024-07-03 09:13:32,750 Tinker Full-Stack App [in /mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/__init__.py:51]
2024-07-16 17:07:19,676 Tinker Full-Stack App [in /mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/__init__.py:51]
2024-07-16 17:10:44,589 Tinker Full-Stack App [in /mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/__init__.py:51]
2024-07-16 17:12:22,806 Tinker Full-Stack App [in /mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/__init__.py:51]
2024-07-16 17:13:19,962 Tinker Full-Stack App [in /mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/__init__.py:51]
2024-07-16 17:18:07,480 Tinker Full-Stack App [in /mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/__init__.py:51]
2024-07-16 17:23:55,919 Exception on /request-password-reset [GET] [in /home/mokeira/.local/lib/python3.8/site-packages/flask/app.py:828]
Traceback (most recent call last):
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/routes.py", line 181, in request_password_reset
    return render_template(
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 149, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/loaders.py", line 125, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 65, in get_source
    return self._get_source_fast(environment, template)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 99, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: request_password_reset.html
2024-07-16 17:23:59,075 Exception on /request-password-reset [GET] [in /home/mokeira/.local/lib/python3.8/site-packages/flask/app.py:828]
Traceback (most recent call last):
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/mnt/c/Users/user/Desktop/Folders/software_development/chapter14_working_with_emails/app/routes.py", line 181, in request_password_reset
    return render_template(
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 149, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/home/mokeira/.local/lib/python3.8/site-packages/jinja2/loaders.py", line 125, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 65, in get_source
    return self._get_source_fast(environment, template)
  File "/home/mokeira/.local/lib/python3.8/site-packages/flask/templating.py", line 99, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: request_password_reset.html
